### ISSUE TITLE
Cannot find IDK credentials

### ISSUE THREAD

**Issue by enzioardesch** [trocotronic/weconnect#18]
Since a few days I am getting the following error. At the time it started the WeConnect app was also having some login problems, but they have been fixed since a few days; yet this error perseveres. 

I have done the following troubleshooting steps:
- Deleted the session and access file. (Running won't recreate these files)  
- Changed the password and S-pin
- Deleted all registered devices in the WeConnect portal
- Checked all possible environments but no permissions that need to be accepted. 

````
[19/07/2023 16:16:17] [API::WARNING] Session file not found
[19/07/2023 16:16:17] [API::WARNING] Access file not found
[19/07/2023 16:16:17] [API::WARNING] Forcing login
[19/07/2023 16:16:17] [urllib3.connectionpool::DEBUG] Starting new HTTPS connection (1): identity.vwgroup.io:443
[19/07/2023 16:16:17] [urllib3.connectionpool::DEBUG] https://identity.vwgroup.io:443 "GET /oidc/v1/authorize?prompt=login&state=czLrxLH8VLRLQTHpk8ZMosYO7mSrf9OBkmC6uWApZ3K&response_type=code+id_token+token&code_challenge_method=s256&scope=openid+profile+mbb+cars+birthdate+nickname+address+phone&code_challenge=41FAbR29_1dDSArQRUuHk9rYrdf2AL3bokSU5htvA7g&redirect_uri=carnet%3A%2F%2Fidentity-kit%2Flogin&client_id=9496332b-ea03-4091-a224-8c746b885068%40apps_vw-dilab_com&nonce=7g05VWidpotfRFrIm-vuEWUaHvlOIWdru5VDkmHZnF6 HTTP/1.1" 302 0
[19/07/2023 16:16:17] [urllib3.connectionpool::DEBUG] https://identity.vwgroup.io:443 "GET /signin-service/v1/signin/9496332b-ea03-4091-a224-8c746b885068@apps_vw-dilab_com?relayState=5c188eccbadfc262d31e62776f55fc5ed58f9360 HTTP/1.1" 200 58724
[19/07/2023 16:16:18] [urllib3.connectionpool::DEBUG] https://identity.vwgroup.io:443 "POST /signin-service/v1/9496332b-ea03-4091-a224-8c746b885068@apps_vw-dilab_com/login/identifier HTTP/1.1" 303 0
[19/07/2023 16:16:18] [urllib3.connectionpool::DEBUG] https://identity.vwgroup.io:443 "GET /signin-service/v1/9496332b-ea03-4091-a224-8c746b885068@apps_vw-dilab_com/login/authenticate?relayState=5c188eccbadfc262d31e62776f55fc5ed58f9360 HTTP/1.1" 200 55772
[19/07/2023 16:16:18] [API::CRITICAL] Raising error msg: Cannot find IDK credentials
Traceback (most recent call last):
  File "/home/deb117581/kmlogpython/NativeAPI.py", line 270, in __get_idk
    idk = json.loads(idk_txt)
  File "/opt/alt/python310/lib64/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/opt/alt/python310/lib64/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/opt/alt/python310/lib64/python3.10/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 5 column 13 (char 1188)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/deb117581/kmlogpython/kmlogscriptarray.py", line 8, in <module>
    vwc.login()
  File "/home/deb117581/kmlogpython/NativeAPI.py", line 290, in login
    return self.__force_login()
  File "/home/deb117581/kmlogpython/NativeAPI.py", line 358, in __force_login
    idk = self.__get_idk(soup)
  File "/home/deb117581/kmlogpython/NativeAPI.py", line 273, in __get_idk
    raise VWError('Cannot find IDK credentials')
NativeAPI.VWError: Cannot find IDK credentials
````

**Comment by TobiasIW** [comment]
Same issue for me.
There seems to be a "," too much in 
`"disabledFeatures": {
                "isFooterEnabled": true,
            },`

If I remove this with:
`idk_txt = idk_txt.replace(": true,", ": \"true\"")`
then it works. This is of course not the solution but I hope it helps ;)

complete json:
`{
            "templateModel": {"clientLegalEntityModel":{"clientId":"9496332b-ea03-4091-a224-8c746b885068@apps_vw-dilab_com","clientAppName":"We Connect App","clientAppDisplayName":"We Connect App","legalEntityInfo":{"name":"Volkswagen","shortName":"VOLKSWAGEN","productName":"Volkswagen ID","theme":"volkswagen_d6","defaultLanguage":"en","termAndConditionsType":"DEFAULT","legalProperties":{"revokeDataContact":"info-datenschutz@volkswagen.de","imprintText":"IMPRINT","countryOfJurisdiction":"DE"}},"legalEntityCode":"volkswagen","imprintTextKey":"imprint.link.text"},"template":"loginAuthenticate","hmac":"a357cc186bc262c8571fda4c48bfe0cd3c06e60fd1f3ad2579e34c29cba82c2f","useClientRendering":true,"titleKey":"title.login.password","title":null,"emailPasswordForm":{"@class":"com.volkswagen.identitykit.signin.domain.model.dto.EmailPasswordForm","email":"my@email.com","password":null},"error":null,"relayState":"ed2347cf6f075f198c1610910dad5a7cf94a6d00","nextButtonDisabled":false,"enableNextButtonAfterSeconds":0,"postAction":"login/authenticate","identifierUrl":"login/identifier"},
            "disabledFeatures": {
                "isFooterEnabled": true,
            },
            "currentLocale": "en",
            "csrf_parameterName": "_csrf",
            "csrf_token": "Adco30Ru0d8nRHgMVYviMqRKf8E2ezMwZnPjqLWZTasckv0SM-ZOvnwIs-4KcUA6MKbWUcEpUvgDTwYdVRLSmdeuLJ94oZsl",
            "userSession": {
                "userId": "",
                "countryOfResidence": ""
            },
            "baseUrl": "https://identity.vwgroup.io",
            "consentBaseUrl": "http://consentapi-service.idk-peu.svc.cluster.local:8080"
        }`

**Comment by trocotronic** [comment]
Fixed in [f75d1a4](https://github.com/trocotronic/weconnect/commit/f75d1a440e8d275d7285d6dc0d9a18590dfeaf37)

Requires `pyyaml`.
